Marge B92 Zephyr updates into new develop branch

serhiiSalamakha opened the pull request:
http://192.168.48.49:8080/browse/MATTER-362

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/4cab17cc82b9ca764ffad3df5a621171f7e6f21d (develop,new_develop) | https://github.com/telink-semi/hal_telink/commit/2f4da732d92cf0a785c43f887c3c5ff7d8bb3e68 (telink_B92) | [telink-semi/hal_telink@4cab17cc..2f4da732](https://github.com/telink-semi/hal_telink/compare/4cab17cc82b9ca764ffad3df5a621171f7e6f21d..2f4da732d92cf0a785c43f887c3c5ff7d8bb3e68) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

s07641069 commented on drivers/adc/adc_b9x.c:
```diff
@@ -390,84 +390,77 @@ static int adc_b91_channel_setup(const struct device *dev,
 
 	/* Init ADC */
 	data->differential = channel_cfg->differential;
-	adc_init(vref_internal_mv, pre_scale, sample_freq);
```
here removed part of code, have we tested impact into B91?

serhiiSalamakha replied:
_**adc_battery_voltage_sample_init**_ and **_adc_gpio_sample_init_** functions call _**adc_init**_.
Yes, it was also tested in B91.

s07641069 commented on drivers/spi/spi_b9x.c:
```diff
@@ -101,55 +111,78 @@ static bool spi_b91_config_cs(const struct device *dev,
 		/* enable cs pin if it is defined and is requested */
 		if ((cs_pin != 0) && (cs_id == config->slave)) {
-			if (b91_config->peripheral_id == PSPI_MODULE) {
-				pspi_set_pin_mux(cs_pin);
```
Is this code line removed by mistake?> pspi_set_pin_mux(cs_pin);

serhiiSalamakha replied:
**_pspi_cs_pin_en(cs_pin)_** function is a wrapper for the  **_pspi_set_pin_mux_**. 
It is not necessary to do it twice.
Also **_hspi_cs_pin_en(cs_pin)_** function is a wrapper for the  **_hspi_set_pin_mux_**.

s07641069 commented on drivers/spi/spi_b9x.c:
```diff
@@ -160,23 +193,18 @@ static void spi_b91_tx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t l
 			tx = 0;
 		}
 		spi_context_update_tx(ctx, 1, 1);
-		while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_TXF_FULL) {
```
why removed?

serhiiSalamakha replied:
_**hal_spi_write**_(spi_write in hal spi driver) contains the _**while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_TXF_FULL)**_ check.

s07641069 commented on drivers/spi/spi_b9x.c:
```diff
@@ -160,23 +193,18 @@ static void spi_b91_tx(uint8_t peripheral_id, struct spi_context *ctx, uint8_t l
 	uint8_t rx = 0;
 
 	for (int i = 0; i < len; i++) {
-		while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_RXF_EMPTY) {
```
why removed?

serhiiSalamakha replied:
_**hal_spi_read**_(spi_read in hal spi driver) contains the _**while (reg_spi_fifo_state(peripheral_id) & FLD_SPI_RXF_EMPTY)**_ check.

s07641069 commented on soc/riscv/riscv-privilege/telink_b9x/Kconfig.defconfig.series:
```diff
@@ -29,7 +29,7 @@ config RISCV_GP
 config NUM_IRQS
 	int
-	default 76
+	default 59
```
why we need this change?

serhiiSalamakha replied:
The **_NUM_IRQS_** was updated.

s07641069 reviewed (approved):
@andriy-bilynskyy could you add your review?
Jenkins tests still in progress.
GitHub CI under fixing.

andriy-bilynskyy commented on .github/workflows/telink-build.yaml:
Could you split this job into 2 files for b91 and b92

serhiiSalamakha replied:
Done

andriy-bilynskyy commented:
Could you update both branches to latest develop. In that case I'll start testing our samples from artifacts.

interfer committed:
drivers: pinctrl: dts: unified pinctrl driver for B9x-series

Pinctrl driver became common for B91 and B92

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
docs: tlsr9528 newline fix

Added newline indentation to fix CI failure

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

s07641069 committed:
soc: Add draft B92 soc support

Create Telink zephyr SoCs hierarchy to avoid code duplication
Create working base sample (run core/blink LED)

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

interfer committed:
drivers: dts: gpio: restyled naming issues

Post-review changes, mainly styling

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
drivers: gpio: dts: modified gpio_b91.c and DTS/DTSI

UART0,1 added, gpio_b91.c modified

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
drivers: dts: gpio: modified gpio driver and boards/b92.dts

GPIO driver will use DTS bindings

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
soc: Add B92 description to boards

Added B92 description to boards

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
drivers: dts: pinctrl: modified B92 gpio definitions

Minor review fixes, modified b92-pinctrl.dtsi, filled dts

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

serhiiSalamakha committed:
drivers: serial: dts: refactor serial driver and DTS to b9x-uart

Refactored serial driver and DTS to b9x-uart

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

interfer committed:
drivers: gpio: dts: changed gpio addresses as per datasheet

Modified adresses to match the datasheet on TLSR9528a

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
drivers: gpio: dts: refactored gpio driver and DTS to b9x-series

From now b91-gpio and b92-gpio become single b9x-gpio driver
as there were minimal changes to make Zephyr driver's API
in gpio_b91.c be used for both SoCs

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

serhiiSalamakha committed:
drivers: hwinfo: update hwinfo driver for B91 and B92

Updated hwinfo_b91 driver to hwinfo_b9x for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

mishadesh committed:
drivers: ieee802154: b9x: Updated IEEE802154 power table

Updated the IEEE802154 power table to match with the new RF driver

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

serhiiSalamakha committed:
drivers: pwm: pinctrl: dts: update i2c driver for B91 and B92

Updated i2c_b91 driver to i2c_b9x for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

interfer committed:
spi: fixed spi_flash sample overlay

Changed spi frequency in DTS overlay
to fix failing to start

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

serhiiSalamakha committed:
drivers: spi: dtsi: update spi driver for B91 and B92

Updated spi_b91 driver to spi_b9x for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
west.yml: Update telink hal version

Updated hal_telink module revision.

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

s07641069 committed:
drivers: gpio: Add B92 GPIO port F support

Enable port E & F for B92
Fix compliance checks & compilation errors after rebase
Update hal_telink module revision.

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

serhiiSalamakha committed:
drivers: soc: west.yml: add changes after updating hal drivers

Updated SOC and SPI driver

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
soc: stop 32k watchdog for B92

Disabled 32k watchdog which causes B92 to reset

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
drivers: flash: pinctrl: dts: update flash driver for B91 and B92

Updated soc_flash_b91 driver to soc_flash_b9x for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

interfer committed:
drivers: entropy: adopted TRNG driver for B92

Added entropy driver for B92, unified
True Random Number Generator driver for B91
and B92 boards to be common B9x.

Signed-off-by: Dmytro Huz <dmytro.huz@telink-semi.com>

mishadesh committed:
soc: riscv: b9x: Introduced B92 WFI fix

Due to silicon bug in B92 platform, the software WFI mode is implemented

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

serhiiSalamakha committed:
drivers: boards: soc: west.yml: update ble hci driver for B91 and B92

Updated ble hci driver for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

mishadesh committed:
drivers: ieee802154: Introduced united driver for Telink B91 & B92

United the mostly common B91 & B92 IEEE802154 drivers

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

mishadesh committed:
drivers: ieee802154: b9x: Removed MCU selector in b9x_stop

Removed MCU core version selection in ieee802154 driver for Telink MCU

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

serhiiSalamakha committed:
drivers: pinctrl: dts: gpio: fix issue with pin function

Update pinctrl driver and DTS for fix issue with pin function

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

interfer committed:
drivers: soc: enable 4wire-spi on B92

Changed default setting for 4wire-spi

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

serhiiSalamakha committed:
drivers: pwm: pinctrl: dts: update pwm driver for B91 and B92

Updated pwm_b91 driver to pwm_b9x for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
soc: west.yml: update sys_init and west.yml

Updated sys_init and west.yml

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

mishadesh committed:
soc: riscv: telink_b9x: Code style and cosmetic changes

Code style and cosmetic changes

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

mishadesh committed:
drivers: ieee802154: b9x: Updated RF power table

Updated RF power table according to a new RF driver

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

mishadesh committed:
West.yml: Updated hal telink

Updated hal Telink because of new PM sriver

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

mishadesh committed:
samples: drivers: spi_flash: Added Telink B91 & B92 boards

Added Telink B91 and B92 boards support to SPI NOR sample

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

serhiiSalamakha committed:
drivers: soc: dts: update for B91 and B92 after rebase with develop

Updated for B91 and B92 after rebase with develop

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
drivers: soc: cleanup Zephyr SOC

Сleanup Zephyr SOC

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

interfer committed:
drivers: gpio: fixed interrupts on buttons

Modified B92 DTS so port D buttons work now.
As per datasheet changed interrupts number 63->46

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
drivers: crypto: modified crypto for B92

Changes for crypto drivers to be used for B92

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

serhiiSalamakha committed:
drivers: pinctrl: dts: gpio: update gpio and pinctrl drivers

Update gpio and pinctrl drivers for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

TelinkKrystian committed:
drivers: usb: Adapt usb driver for B91 and B92

Update usb_dc_b91 driver to usb_dc_b9x for B91 and B92.

Signed-off-by: Krystian Jankowski <krystian.jankowski@telink-semi.com>

serhiiSalamakha committed:
drivers: adc: dtsi: update adc driver for B91 and B92

Updated adc_b91 driver to adc_b9x for B91 and B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
west.yml: update west.yml after updated pm for B92

Updated west.yml

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
drivers: soc: dtsi: fix problems after rebase B92 branch with develop

Fixed problems after rebase B92 branch with develop

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
drivers: yaml: split telink-build.yaml

- Split the telink-build.yaml file
- Fixed usb_dc_b9x.c after rebasing with develop

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

mishadesh committed:
drivers: gpio: gpio_b9x: Fixed wakeup registers address

Fixed wrong telink B92 gpio wakeup address

Signed-off-by: Misha Tkachenko <misha.tkachenko@telink-semi.com>

serhiiSalamakha committed:
west.yml: update west.yml

Updated west.yml

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

serhiiSalamakha committed:
dts: drivers: yaml: extend GitHub CI and add retention for B92

Extended GitHub CI to build all B92 samples and added retection for B92

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

s07641069 committed:
telink-build.yaml: Need to use correct BOOT_MAX_IMG_SECTORS for B92.

For B91 it set in mcuboot board config file.

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

serhiiSalamakha committed:
west.yml: update west.yml

Updated west.yml

Signed-off-by: Serhii Salamakha <serhii.salamakha@gmail.com>

interfer committed:
drivers: watchdog: telink: watchdog driver for B9x-series

Added watchdog basic functionality.
Signed-off-by: Dmytro Huz <diman1436@gmail.com>
